src/qtism/runtime/tests/AssessmentTestSession.php

```diff
@@ -2438,17 +2440,21 @@ protected function nextRouteItem($ignoreBranchings = false, $ignorePreConditions
             }
 
             // Preconditions on target?
-            if ($ignorePreConditions === false && $route->valid() === true && ($preConditions = $route->current()->getPreConditions()) && count($preConditions) > 0 && $this->mustApplyPreConditions() === true) {
+            if ($ignorePreConditions === false && $route->valid() === true && ($preConditions = $route->current()->getEffectivePreConditions()) && count($preConditions) > 0 && $this->mustApplyPreConditions() === true) {
+                $preConditionFailed = false;
+
```

Comment:
suggestion: We may use this change as an opportunity for moving the side effects (i.e. the var assignment) out of the conditional statement.

```suggestion
            $hasPreContitions = false;
            $preConditions = [];
            
            if ($ignorePreConditions === false && $route->valid()) {
                $preConditions = $route->current()->getEffectivePreConditions();
                $hasPreConditions = count($preConditions) > 0;
            }

            if ($hasPreConditions && $this->mustApplyPreConditions()) {
                $preConditionFailed = false;
```

Reply:
I want to avoid change it too much as is critical lib and I am getting underbudget for the ticket. So I will skip this changes for the moment if you do not mind :)

Reply:
ok, no problem :ok_hand: